gridded_etl_tools/utils/publish.py

```diff
@@ -793,35 +793,147 @@ def check_written_value(
 
     def get_original_ds(self, coords: dict[Any]) -> xr.Dataset:
         """
-        Retrieve the original dataset that corresponds to the given coordinates.
+        Retrieve the original dataset that corresponds to the given coordinates by conducting a consecutive series of
+        binary searches for files matching each coordinate.
 
         Parameters
         ----------
         coords
-            The coordinates to used to locate the original dataset.
+            Randomly selected coordinates from the production dataset
+            used to locate the corresponding original dataset.
 
         Returns
         ----------
         orig_ds
-            The original dataset, unformatted
+            The original dataset, formatted minimally to allow comparison with the production dataset
         """
-        # Randomly select an original dataset
-        raw_ds, orig_file_path = self.binary_search_for_file(coords[self.time_dim], time_dim=self.time_dim)
+        # Apply any custom filters to the input files. Returns all input files if no custom filter defined.
+        possible_files = self.custom_file_filter(list(self.input_files()))
```

Comment:
Does this change between each coordinate check? If not, we probably want this and the other filters in this function to happen before iterating through the coordinates.